src/SmartFormat/Utilities/PluralRules.cs

```diff
@@ -217,44 +231,44 @@ public static class PluralRules
         if (pluralWordsCount == 2) return value < 2 ? 0 : 1;
 
         if (pluralWordsCount == 3) return GetWordsCount3Value(value);
-
+        
         if (pluralWordsCount == 4) return GetWordsCount4Value(value);
 
         return -1;
     }; // DualFromZeroToTwo: one (n == 0..2 fractionate and n != 2), other
 
     private static int GetWordsCount3Value(decimal n)
     {
-        return n switch {
-            0 => 0,
+        return Math.Abs(n) switch
+        {
+            <=0 => 0,
             > 0 and < 2 => 1,
-            > 2 => 2,
-            _ => -1
+            _ => 2
         };
     }
 
     private static int GetWordsCount4Value(decimal n)
     {
-        return n switch {
-            < 0 => 0,
-            0 => 1,
-            > 0 and < 2 => 2,
-            > 2 => 3,
-            _ => -1
+        return Math.Abs(n) switch
+        {
+            <=0 => 0,
+            > 0 and < 2 => 1,
+            >= 2 and < 3 => 2,
+            _ => 3
         };
```

Comment:
@karljj1 While going through the code once again:
[Since the very beginning](https://github.com/axuno/SmartFormat/blob/1f3cd59f43dbbfa2d5e983357d9f1adb9be167b3/src/SmartFormat/Utilities/PluralRules.cs), `PluralRules` are dealing with `decimal`, while comparisons are integer based. As soon as decimal is not an integer, the index for "other" is returned. See also the `Between(...)` method. Can you figure out, what was the intention behind?
If this is the concept, these range comparisons break the concept, don't they?

Reply:
Im not sure I understand. The Between method uses decimals. Maybe Im misunderstanding you?

Reply:
Yes, ` Between` uses decimal, but checks that there is no faction (like 1.0000001). That's what I mean: The comparison there (and in other methods as well) expects that there's no fraction. The range comparison includes fractions and is inconsistent with the other comparisons in this respect, isn't it?
Frankly I don't understand why this kind of implementation was chosen, and I wonder whether there's something I'm just missing.

Reply:
Which range comparison are you referring to that uses fractions?

Reply:
I'm not the author of `PluralRules`, and I think there is an inconsistency. I can't imagine the reason for it.

See these examples:
```C#
// Never include fractional numbers to return true
Assert.That(PluralRules.BetweenWithoutFraction(1.2M, 1M, 2M), Is.False);
Assert.That(PluralRules.BetweenWithoutFraction(1.0M, 1M, 2M), Is.True); // whole number

// Never include fractional numbers
Assert.That(PluralRules.Breton(1M, default), Is.EqualTo(1));
Assert.That(PluralRules.Breton(1.1M, default), Is.EqualTo(5)); // default, not a whole number

// Always include fractional numbers with DualFromZeroToTwo
// Why only here?
Assert.That(PluralRules.GetWordsCount4Value(1M), Is.EqualTo(1));
Assert.That(PluralRules.GetWordsCount4Value(1.1M), Is.EqualTo(1)); // same result for number with fraction
```
What I'm trying to say: Range comparisons for a value are only used with `PluralRules.DualFromZeroToTwo`. This is not consistent or must have a good reason. So what is it?

My guess is, that range comparisons would be the right way for `decimal` (!) values. Why should we use `decimal`, if we compare for equality with whole numbers? Then we could use `long` as well.

Reply:
Yeah, I don't know why it's this way. There is some mention in the CLDR that the _many_ option can also be used for fractions https://cldr.unicode.org/index/cldr-spec/plural-rules

> Fractions
> 
> In some languages, fractions require a separate category. For example, Russian 'other' in the example above. In some languages, they all in a single category with some integers, and in some languages they are in multiple categories. In any case, they also need to be examined to make sure that there are sufficial minimal pairs.
>
> Fractions are often a bit tricky to determine: languages have very different behavior for them. In some languages the fraction is ignored (when selecting the category), in some languages the final digits of the fraction are important, in some languages a number changes category just if there are visible trailing zeros. Make sure to try out a range of fractions to make sure how the numbers behave: values like 1 vs 1.0 may behave differently, as may numbers like 1.1 vs 1.2 vs 1.21, and so on.

Maybe the original used an older version of the CLRD? 

> Fractional support in plurals is new in CLDR 24. Because the fractions didn't work before, the changes in categories from 23 to 24 should not cause an issue for implementations.

Reply:
Thanks a lot, Karl, with your hint to Fraction support the scales fell from my eyes!  ½, 1 ½ sure!
I'll revert to v3.3.0 code, but include 2 as the minimum plural value. Then this is just a bug fix with no breaking changes.